### PR TITLE
README: edit intro and build instructions. Added guix container info

### PR DESCRIPTION
**Pull request by pjotrp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,25 +2,25 @@
...
-Each node is represented by a byte array into which we write variable length integers to represent, 1) the node sequence, 2) its edges, and 3) the paths crossing the node.
+Careful encoding of graph entities allows `odgi` to efficiently compute and transform [pangenomes](https://pangenome.github.io/) with minimal overheads.  `odgi` implements a dynamic data structure that can be updated on the fly. This contrasts with _static_ data structures, such as [xg](https://github.com/vgteam/xg).
 
-The edges and path steps are recorded relativistically, as deltas between the current node id and the target node id, where the node id corresponds to the rank in the global array of nodes.
-Graphs built from biological data sets tend to have local partial order, and when sorted the stored deltas will tend to be small.
-This allows them to be compressed with a variable length integer representation, resulting in a small in-memory footprint at the cost of packing and unpacking. 
+`odgi` implements dynamic [GBWT](https://github.com/jltsiren/gbwt) as a byte-packed version of the graph.
```
Comment (ekg) [inline README.md:11]: n.b. It doesn't implement a dynamic GBWT. It uses some ideas from the implementation of dynamic GBWT.

File: README.md [hunk 2]
```diff
@@ -2,25 +2,25 @@
...
+Each node is represented by a `byte array` into which variable length integers are written to represent: 1) node sequences, 2) edges, and 3) paths through nodes.
```
Comment (ekg) [inline README.md:12]: This is also not correct, we should sync an update I made to the docs that's in the paper draft

File: README.md (hunk 1 above)
Comment (pjotrp) [reply README.md]: I removed the offending line

File: README.md (hunk 2 above)
Comment (pjotrp) [reply README.md]: Removed

**Comment by pjotrp** [comment]
We can add more information on the method later, or simply refer to code/paper.

**Pull request merged**